includes/Products.php

```diff
@@ -651,8 +651,34 @@ public static function get_product_color( \WC_Product $product ) {
 	 */
 	public static function get_product_size_attribute( \WC_Product $product ) {
 
-		// TODO: implement the validations
-		return $product->get_meta( self::SIZE_ATTRIBUTE_META_KEY );
+		if ( $product->is_type( 'variation' ) ) {
+
+			// get the attribute from the parent
+			$parent_product = wc_get_product( $product->get_parent_id() );
+
+			return self::get_product_size_attribute( $parent_product );
```

Comment:
same as the others - avoid multiple returns in getter methods - instead redefining `$product` and continuing with the logic below